bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/WhileMatchFilterAdapter.java

```diff
@@ -75,6 +75,9 @@ public WhileMatchFilterAdapter(FilterAdapter subFilterAdapter) {
    */
   @Override
   public RowFilter adapt(FilterAdapterContext context, WhileMatchFilter filter) throws IOException {
+    checkArgument(
+        context.getNumberOfWhileMatchFilters() == 0,
```

Comment:
The intention is keeping track of all WMFs been added. I would say checking size for enforcing size limit makes more sense in this case.